src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java

```diff
@@ -43,4 +60,90 @@
+   */
+  @Override
+  public void fulfilmentRequestBySMS(SMSFulfilmentRequestDTO requestBodyDTO) throws CTPException {
+    UUID caseId = requestBodyDTO.getCaseId();
+
+    // Read case from firestore
+    Optional<CollectionCase> caseDetails = dataRepo.readCollectionCase(caseId.toString());
+    if (caseDetails.isEmpty()) {
+      String errorMessage = "Case not found: " + caseId;
+      log.info(errorMessage);
+      throw new CTPException(Fault.RESOURCE_NOT_FOUND, errorMessage);
+    }
+
+    // Attempt to find the requested product
+    Product product =
+        findProduct(caseDetails.get(), requestBodyDTO.getFulfilmentCode(), DeliveryChannel.SMS);
+    if (product == null) {
+      log.info("fulfilmentRequestBySMS can't find compatible product");
+      throw new CTPException(Fault.BAD_REQUEST, "Compatible product cannot be found");
+    }
+
+    // Build and send a fulfilment request event
+    FulfilmentRequest fulfilmentRequestedPayload =
+        createFulfilmentRequestPayload(product, requestBodyDTO.getTelNo(), caseDetails.get());
+    publisher.sendEvent(
+        EventType.FULFILMENT_REQUESTED,
+        Source.RESPONDENT_HOME,
+        Channel.RH,
+        fulfilmentRequestedPayload);
+  }
+
+  // Search the ProductReference for the specified product
+  private Product findProduct(
+      CollectionCase caseDetails, String fulfilmentCode, Product.DeliveryChannel deliveryChannel)
+      throws CTPException {
+
+    Region region = Region.valueOf(caseDetails.getAddress().getRegion());
+    CaseType caseType = CaseType.valueOf(caseDetails.getAddress().getAddressType());
+
+    log.with("region", region)
+        .with("caseType", caseType)
+        .with("deliveryChannel", deliveryChannel)
+        .with("fulfilmentCode", fulfilmentCode)
+        .debug("Attempting to find product.");
+
+    // Build search criteria base on the cases details and the requested fulfilmentCode
+    Product searchCriteria = new Product();
+    searchCriteria.setRequestChannels(Arrays.asList(Product.RequestChannel.RH));
+    searchCriteria.setRegions(Arrays.asList(region));
+    searchCriteria.setCaseType(caseType);
+    searchCriteria.setDeliveryChannel(deliveryChannel);
+    searchCriteria.setFulfilmentCode(fulfilmentCode);
+
+    // Attempt to find matching product
+    List<Product> products = productReference.searchProducts(searchCriteria);
+    if (products.size() == 0) {
+      return null;
+    }
+
+    return products.get(0);
+  }
+
+  private FulfilmentRequest createFulfilmentRequestPayload(
+      Product product, String telephoneNumber, CollectionCase caseDetails) {
+    // Create the event payload request
+    FulfilmentRequest fulfilmentRequest = new FulfilmentRequest();
+    fulfilmentRequest.setCaseId(caseDetails.getId());
+    if (product.getCaseType().equals(Product.CaseType.HI)) {
+      fulfilmentRequest.setIndividualCaseId(UUID.randomUUID().toString());
+    }
+
+    fulfilmentRequest.setFulfilmentCode(product.getFulfilmentCode());
```

Comment:
Don't set Address and _only_ set tel no in contact

Reply:
ok.
Code is no longer setting address or generic contact details. Only setting phone number in contact details.